File: py/IEC62325/MarketOperations/ReferenceData/FuelRegion.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106 on Thu Dec 28 12:28:33 2023
from typing import Optional

from IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from IEC61970.Base.Domain.DateTime import DateTime
from IEC62325.MarketOperations.ReferenceData.RegisteredGenerator import RegisteredGenerator


class FuelRegion(IdentifiedObject):
    """
    Indication of region for fuel inventory purposes.
    @updated 28-Dec-2023 12:14:44 PM
    """

    def __init__(self) -> None:
        super().__init__()
        self.fuel_region_typeOptional[str] = ""
        self.last_modifiedOptional[DateTime] = DateTime()
        self.registered_generatorOptional[RegisteredGenerator] = RegisteredGenerator()

File: java/IEC62325/MarketOperations/ReferenceData/FuelRegion.java
package IEC62325.MarketOperations.ReferenceData;

import IEC61970.Base.Domain.String;
import IEC61970.Base.Domain.DateTime;
import IEC61970.Base.Core.IdentifiedObject;

/**
 * Indication of region for fuel inventory purposes.
 * @updated 28-Dec-2023 12:14:44 PM
 */
public class FuelRegion extends IdentifiedObject {

	/**
	 * The type of fuel region
	 */
	public String fuelRegionType;
	/**
	 * Time of last update
	 */
	public DateTime lastModified;
	public RegisteredGenerator RegisteredGenerator;

	public FuelRegion(){

	}
}//end FuelRegion